### PR TITLE
Classifier - TESS Light Curve Viewer - Zoom buttons

### LINKED ISSUE #237: TESS LCV - viewer should listen to Zoom In/Out buttons on
## Required Feature

Package: lib-classifier
Project: Light Curve Viewer / Planet Hunters TESS

The TESS Light Curve Viewer needs to be able to clicks on the Zoom In/Zoom Out button on the Classifier toolbar. At the moment, the TESS LCV can zoom in/out of the light curve by hooking up the D3 code to listen to mouse scrolls.

The main catch here is that the D3 code in `LightCurveViewer.js` keeps track of, and controls, its own zoom levels (it's the `.zoom` controller attached to the `InterfaceLayer`; it has its own `transform` data). While it's entirely possible to programmatically change the zoom levels, the LightCurveViewer needs to know when a user clicked on the Zoom In or Zoom out buttons.

Hence, the most likely solution: the LightCurveViewer needs to be able to pass an `onZoom` or `customZoomHandler` callback function to the SubjectViewerStore (which knows when the Zoom In and Zoom Out buttons are clicked).

**To Do list:**
- [ ] modify SubjectViewerStore to accept an `onZoom` callback function; this function will trigger appropriately when `SubjectViewerStore.zoomIn` and `SubjectViewerStore.zoomOut` are called.
  - [ ] Optional: also have a `setZoom` function? Might be handy.
  - To consider: what happens when the user clicks the Reset button? Add an `onReset` callback?
- [ ] hook up LightCurveViewer.js to register said callback functions to the SubjectViewerStore
- [ ] When LightCurveViewers's onZoom callback function is triggered, programmatically update the zoom level of the LCV. Reference: https://bl.ocks.org/mbostock/7ec977c95910dd026812

**Templates/Starting points:**
- Study how `DrawingStore` uses `.volatile`.

### PR DESCRIPTION
**Pull request by shaunanoordin**
## PR Overview
Package: `lib-classifier`
Closes #237 

The goal of this PR is to hook up the LCV with the Classifier's Toolbar
- [x] The Zoom buttons on the Classifier toolbar should allow users to zoom in/out of the Light Curve Viewer.
- [x] The Reset Image button on the Classifier toolbar should reset the LCV to its default zoom and view position.

Simple in theory, but there are two main hurdles:
- we'll be modifying the Subject Viewer Store with an "onZoom" callback function, or something similar.
- The D3 code in LightCurveViewer.js needs to programmatically update the Transform of the Zoom object: https://bl.ocks.org/mbostock/7ec977c95910dd026812

### Status
~~WIP~~
~~This branch is based off #219 and should not be reviewed or merged until that one is.~~
~~Development for this PR is now complete. However, it's _probably_ advisable to not review this until #219 is merged, and it should _definitely_ not be merged until 219 is. (2018.11.08)~~

Ready for review 👍 (2011.11.12)

### CODE REVIEW

**Comment by shaunanoordin** [comment]
## PR Update
My proposal for the `onZoom` event hook _(is that even the correct term?)_ is as follows:

classifier.store. **SubjectViewer.onZoom** : `function (type, n)`
Specifies a callback function that triggers when a Zoom In, Zoom Out, or Zoom To action is triggered via the Classifier's ImageToolbar component. This is useful when a specialised subject viewer handles its own zoom controls, instead of listening to an absolute "current zoom value" listed in the Subject Viewer store.

Callback function is structured as:

callback : `function (type, n)`
- `type` is a string value specifying type of action, either 'zoomin', 'zoomout', or 'zoomto'
- `n` is the zoom value
  - for 'zoomin', `n` is a positive value
  - for 'zoomout', `n` is a negative value
  - for 'zoomto', `n` is the new zoom value that's being set. e.g. on the "Reset Image" action, the value of `n` would be `1.0`

In practice, the `n` value for 'zoomin' and 'zoomout' would be ignored, as it's sufficient for the custom zoom handler to understand what the attempted action was from the `type`. 🤷‍♂️ (i.e. when clicking the Zoom In button on the Classifier's ImageToolbar, `n` would always be `+1` and never, say, `+10` for a SUPER ZOOM action.)

**Comment by shaunanoordin** [comment]
New feature:
- the Reset Image button on the Classifier toolbar now resets the view on the light curve.

**Comment by shaunanoordin** [comment]
## Update
PR #219 has been merged and this PR is now ready for review 👍

**Comment by shaunanoordin** [comment]
This PR has been updated to resolve merge conflicts after the merge of #261

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewer.js [hunk 1]
```diff
@@ -138,6 +149,25 @@ class LightCurveViewer extends Component {
...
+    const ZOOM_OUT_VALUE = 0.8
+    const ZOOMING_TIME = 100  // milliseconds
+    
+    if (type === 'zoomin') {
+      this.zoom.scaleBy(this.d3interfaceLayer.transition().duration(ZOOMING_TIME), ZOOM_IN_VALUE)
+    } else if (type === 'zoomout') {
+      this.zoom.scaleBy(this.d3interfaceLayer.transition().duration(ZOOMING_TIME), ZOOM_OUT_VALUE)
+    } else if (type === 'zoomto') {
+      this.zoom.scaleTo(this.d3interfaceLayer.transition().duration(ZOOMING_TIME), n)
+    }
```
[the remark this answers is not in the document]
Comment (rogerhutchings) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewer.js]: Would you mind addressing ^ @shaunanoordin? I'd be happy to merge this today.

**Comment by shaunanoordin** [comment]
Thanks @srallen ! Good suggestion on the Zoom handler; I'll get this implemented shortly. Also, I'll rename `n` to the more easily understandable `zoomValue` - sorry for that, D3 uses `k` to indicate scale (skale?) with `k=1.0` being 100% zoom, but I didn't know how to communicate that properly. And somehow I thought `n` made more sense. Err, to be fair, it was pretty late when I coded that.

@rogerhutchings thanks for the prod, I'll get this fixed shortly before I get distracted by bug hunts again. Away!

**Pull request merged**